src/libstd/sys/unix/fs.rs

```diff
@@ -510,6 +510,19 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
     Ok(())
 }
 
+pub fn remove_dir_all(path: &Path) -> io::Result<()> {
+    for child in try!(readdir(path)) {
+        let child = try!(child).path();
+        let stat = try!(lstat(&*child));
+        if stat.file_type().is_dir() {
```

Comment:
Could this actually switch to using `child.file_type()`? That'll avoid an extra `lstat` if possible and I believe it returns the same form of information.